crates/web5/src/credentials/status_list_credential.rs

```diff
@@ -0,0 +1,560 @@
+        }
+
+        // Get the bit index
+        let index = status.status_list_index.parse::<usize>().map_err(|_| {
+            Web5Error::Parameter(format!(
+                "invalid status list index: {}",
+                status.status_list_index
+            ))
+        })?;
+
+        let encoded_list = match self
+            .base
+            .credential_subject
+            .additional_properties
+            .as_ref()
+            .and_then(|props| props.properties.get("encodedList"))
+            .and_then(|value| {
+                if let JsonValue::String(s) = value {
+                    Some(s.as_str())
+                } else {
+                    None
+                }
+            }) {
+            Some(el) => el,
+            None => {
+                return Err(Web5Error::Parameter(
+                    "invalid or missing encodedList".to_string(),
+                ))
+            }
+        };
+
+        // Check the bit in the encoded list
+        Self::get_bit(encoded_list, index)
+    }
+
+    /// Extracts status list indexes from a vector of verifiable credentials that match the specified status purpose.
+    ///
+    /// # Arguments
+    /// * `status_purpose` - The status purpose to match.
+    /// * `credentials` - A vector of `VerifiableCredential` objects.
+    ///
+    /// # Returns
+    /// A `Result` containing a vector of `usize` indexes, or an error if a credential is missing
+    fn get_status_list_indexes(
+        status_purpose: &str,
+        credentials: Vec<VerifiableCredential>,
+    ) -> Result<Vec<usize>> {
+        let mut status_list_indexes = Vec::new();
+
+        for vc in credentials {
+            let status_list_entry = vc.credential_status.as_ref().ok_or_else(|| {
+                Web5Error::Parameter("no credential status found in credential".to_string())
+            })?;
+
+            if status_list_entry.status_purpose != *status_purpose {
+                return Err(Web5Error::Parameter("status purpose mismatch".to_string()));
+            }
+
+            let index = status_list_entry
+                .status_list_index
+                .parse::<usize>()
+                .map_err(|_| {
+                    Web5Error::Parameter(format!(
+                        "invalid status list index: {}",
+                        status_list_entry.status_list_index
+                    ))
+                })?;
+
+            status_list_indexes.push(index);
+        }
+
+        Ok(status_list_indexes)
+    }
+
+    /// Generates a compressed, base64-encoded bitstring from a list of status list indexes.
+    ///
+    /// # Arguments
+    /// * `status_list_indexes` - A vector of indexes to set in the bitstring.
+    ///
+    /// # Returns
+    /// A `Result` containing the compressed, base64-encoded bitstring, or an error if an index is out of range.
+    fn bitstring_generation(status_list_indexes: Vec<usize>) -> Result<String> {
```

Comment:
nit: don't need to take ownership. might avoid unnecessary cloning
```suggestion
    fn bitstring_generation(status_list_indexes: &[usize]) -> Result<String> {
```